pkg/controllers/plugin/fscache.go

```diff
@@ -70,8 +68,44 @@ func (c *FSCache) Sync(namespace string, cache plugincontroller.UIPluginCache) e
 	return nil
 }
 
+// SyncWithIndex syncs up entries in the filesystem cache with the index's entries
+// Entries that aren't in the index, but present in the filesystem cache are deleted
+func (c FSCache) SyncWithIndex(index *SafeIndex, fsCacheFiles []string) error {
```

Comment:
Relocated this to be with the FS Cache stuff as most of the logic is related to the FS Cache and it didn't make sense to have it in index.go